go/cmd/dolt/cli/arg_parser_helpers.go

```diff
@@ -310,6 +311,12 @@ func CreateLogArgParser() *argparser.ArgParser {
 	return ap
 }
 
+func CreatePatchArgParser() *argparser.ArgParser {
+	ap := argparser.NewArgParser()
+	ap.SupportsFlag(CachedFlag, "c", "Show only the unstaged data changes.")
```

Comment:
There's another comment below about the `--cached` flag being documented backwards. Similar change needed here, too: 
```suggestion
	ap.SupportsFlag(CachedFlag, "c", "Show only the staged data changes.")
```